scripts/add_extra_components.py

```diff
@@ -222,6 +223,128 @@ def attach_stores(n, costs, config):
         )
 
 
+def attach_cooking_technologies(n, cooking_costs, config):
+    if config["clean_cooking"]["enable"] != True:
+        return
+
+    carriers = config["clean_cooking"]["fuel"]
+    cooking_costs = cooking_costs
+    buses_i = n.buses.index
+    buses_i = [
+        bus for bus in buses_i if not (bus.endswith("battery") or bus.endswith("H2"))
+    ]
+
+    bus_sub_dict = {k: n.buses[k].values for k in ["x", "y", "country"]}
+    for key in bus_sub_dict:
+        bus_sub_dict[key] = bus_sub_dict[key][: len(buses_i)]
+
+    cooking_buses_i = None
+    if "heat" in carriers:
+        cooking_buses_i = n.madd(
+            "Bus", [bus + " cooking" for bus in buses_i], carrier="heat", **bus_sub_dict
+        )
+
+    for fuel in carriers:
+        if fuel == "heat":
+            continue
+
+        if fuel == "AC" and "AC" in n.buses.carrier.values:
+            fuel_buses_i = n.buses.index[n.buses.carrier == "AC"]
+        else:
+            fuel_buses_i = n.madd(
+                "Bus",
+                [bus + f" {fuel}" for bus in buses_i],
+                carrier=fuel,
+                **bus_sub_dict,
+            )
+
+            n.madd(
+                "Store",
+                fuel_buses_i,
+                bus=fuel_buses_i,
+                carrier=fuel,
+                e_cyclic=fuel == "AC",
+                e_initial=cooking_costs.at[fuel, "e_initial"],
+                capital_cost=cooking_costs.at[fuel, "capital_cost"],
+                marginal_cost=cooking_costs.at[fuel, "marginal_cost"],
+            )
+
+        if cooking_buses_i is not None:
+            min_length = min(len(fuel_buses_i), len(cooking_buses_i))
+            fuel_buses_i = fuel_buses_i[:min_length]
+            cooking_buses_i = cooking_buses_i[:min_length]
+
+            n.madd(
+                "Link",
+                fuel_buses_i + " stove",
+                bus0=fuel_buses_i,
+                bus1=cooking_buses_i,
+                carrier=fuel,
+                efficiency=cooking_costs.at[f"{fuel} stove", "efficiency"],
+                capital_cost=cooking_costs.at[f"{fuel} stove", "capital_cost"],
+                marginal_cost=cooking_costs.at[f"{fuel} stove", "marginal_cost"],
+                p_nom=cooking_costs.at[f"{fuel} stove", "p_nom"],
+            )
+
+
+def attach_cooking_load(n, config, demand_cooking):
+    if config["clean_cooking"]["enable"] != True:
+        return
+
+    demand_df = read_csv_nafix(demand_cooking, index_col=0, parse_dates=True)
+    cooking_bus = n.buses.loc[n.buses.index.str.endswith("cooking")].index
+    n.madd("Load", cooking_bus, bus=cooking_bus, p_set=demand_df)
```

Comment:
Would probably also add the cooking demand like in the sector coupled model: 
https://github.com/PyPSA/pypsa-eur/blob/b49895a0411533a47803e176ac149e6ea7bba9c4/rules/build_sector.smk#L933-L1066

One rule creates a csv/nc file for the cooking demand & then the prepare-network rule is loading the data into the network, if cooking is enabled https://github.com/PyPSA/pypsa-eur/blob/b49895a0411533a47803e176ac149e6ea7bba9c4/rules/build_sector.smk#L1058-L1062